### ISSUE TITLE
Upgrade to AlertManager v0.25.0

### ISSUE THREAD

**Issue by alexandreLamarre** [rancher/opni#945]
Upgrading to v0.25.0 api types & upgrade opni alertmanager command.

Changelog for 0.25.0:
* [CHANGE] Change the default `parse_mode` value from `MarkdownV2` to `HTML` for Telegram. #2981
* [CHANGE] Make `api_url` field optional for Telegram. #2981
* [CHANGE] Use CanonicalMIMEHeaderKey instead of TitleCasing for email headers. #3080
* [CHANGE] Reduce the number of notification logs broadcasted between peers by expiring them after (2 * repeat interval). #2982
* [FEATURE] Add `proxy_url` support for OAuth2 in HTTP client configuration. #3010
* [FEATURE] Reload TLS certificate and key from disk when updated. #3168
* [FEATURE] Add Discord integration. #2948
* [FEATURE] Add Webex integration. #3132
* [ENHANCEMENT] Add `--web.systemd-socket` flag to systemd socket activation listeners instead of port listeners (Linux only). #3140
* [ENHANCEMENT] Add `enable_http2` support in HTTP client configuration. #3010
* [ENHANCEMENT] Add `min_version` support to select the minimum TLS version in HTTP client configuration. #3010
* [ENHANCEMENT] Add `max_version` support to select the maximum TLS version in HTTP client configuration. #3168
* [ENHANCEMENT] Emit warning logs when truncating messages in notifications. #3145
* [ENHANCEMENT] Add `--data.maintenance-interval` flag to define the interval between the garbage collection and snapshotting to disk of the silences and the notification logs. #2849
* [ENHANCEMENT] Support HEAD method for the `/-/healty` and `/-/ready` endpoints. #3039
* [ENHANCEMENT] Truncate messages with the `…` ellipsis character instead of the 3-dots string `...`. #3072
* [ENHANCEMENT] Add support for reading global and local SMTP passwords from files. #3038
* [ENHANCEMENT] Add Location support to time intervals. #2782
* [ENHANCEMENT] UI: Add 'Link' button to alerts in list. #2880
* [ENHANCEMENT] Add the `source` field to the PagerDuty configuration. #3106
* [ENHANCEMENT] Add support for reading PagerDuty routing and service keys from files. #3107
* [ENHANCEMENT] Log response details when notifications fail for Webhooks, Pushover and VictorOps. #3103
* [ENHANCEMENT] UI: Allow to choose the first day of the week as Sunday or Monday. #3093
* [ENHANCEMENT] Add support for reading VictorOps API key from file. #3111
* [ENHANCEMENT] Support templating for Opsgenie's responder type. #3060
* [BUGFIX] Fail configuration loading if `api_key` and `api_key_file` are defined at the same time. #2910
* [BUGFIX] Fix the `alertmanager_alerts` metric to avoid counting resolved alerts as active. Also added a new `alertmanager_marked_alerts` metric that retain the old behavior. #2943
* [BUGFIX] Trim contents of Slack API URLs when reading from files. #2929
* [BUGFIX] amtool: Avoid panic when the label value matcher is empty. #2968
* [BUGFIX] Fail configuration loading if `api_url` is empty for OpsGenie. #2910
* [BUGFIX] Fix email template for resolved notifications. #3166
* [BUGFIX] Use the HTML template engine when the parse mode is HTML for Telegram. #3183

**Comment by alexandreLamarre** [comment]
Upgrading the code results in a dependency error : 
```
Running target: Build
exec: go "build" "-ldflags" "-w -s -X github.com/rancher/opni/pkg/versions.Version=unversioned -X github.com/rancher/opni/pkg/versions.BuildTime=2022-12-29T22:40:57Z" "-trimpath" "-o" "./bin/" "./cmd/..." "./internal/cmd/..." "./plugins/..."
# github.com/thanos-io/thanos/pkg/httpconfig
../../../go/pkg/mod/github.com/thanos-io/thanos@v0.19.1-0.20211208205607-d1acaea2a11a/pkg/httpconfig/http.go:172:73: not enough arguments in call to config_util.NewTLSRoundTripper
        have (*tls.Config, string, func(tlsConfig *tls.Config) ("net/http".RoundTripper, error))
        want (*tls.Config, string, string, string, func(*tls.Config) ("net/http".RoundTripper, error))
# github.com/weaveworks/common/server
../../../go/pkg/mod/github.com/weaveworks/common@v0.0.0-20220927162600-d0a1a93a15be/server/server.go:263:51: undefined: web.TLSStruct
../../../go/pkg/mod/github.com/weaveworks/common@v0.0.0-20220927162600-d0a1a93a15be/server/server.go:278:51: undefined: web.TLSStruct
Error: running "go build -ldflags -w -s -X github.com/rancher/opni/pkg/versions.Version=unversioned -X github.com/rancher/opni/pkg/versions.BuildTime=2022-12-29T22:40:57Z -trimpath -o ./bin/ ./cmd/... ./internal/cmd/... ./plugins/..." failed with exit code 2
```

**Comment by alexandreLamarre** [comment]
Fixed the thanos-io dependency using a somewhat ad-hoc replace block in the go.mod : 
```
// AlertManager replacements
replace (
	github.com/prometheus/alertmanager/alertmanager025 => github.com/prometheus/alertmanager v0.25.0
	github.com/prometheus/alertmanager/alertmanager025/api025 => github.com/prometheus/alertmanager/api v0.25.0
	// ! Freeze client golang to v1.13
	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.13.0

	github.com/prometheus/client_golang/client_golang114 => github.com/prometheus/client_golang v1.14.0
	// ! Freeze prometheus common to v0.37
	github.com/prometheus/common => github.com/prometheus/common v0.37.0

	github.com/prometheus/common/common038 => github.com/prometheus/common v0.38.0
```

the weavescope dependency issue persists

**Comment by alexandreLamarre** [comment]
Ultimately I think this will be blocked by:
1. cortex-project/cortex upgrading to alertmanager 0.25.0
2. upgrading our version of cortex to match the upstream version

**Comment by alexandreLamarre** [comment]
closed with #942
